bot.py

```diff
@@ -139,60 +155,92 @@ def next_in_queue(self, _):
             self.voice_client.stop()
             self.voice_client.play(self.song_queue.get(), after=self.next_in_queue)
 
+    async def get_voice_channel(self, message):
+        """
+        Get voice channel for message author. Complain if author is not in a channel
+        """
+        assert isinstance(message, discord.Message)
+        if message.author.voice is None:
+            await message.channel.send("You are not connected to a voice channel!")
+            return None
+        voice_channel = message.author.voice.channel
+        return voice_channel
+
+    async def get_voice_client(self, message):
+        """
+        Get voice client for message author. Complain if author is not in a channel,
+        connect to author voice client if not yet connected
+        """
+        assert isinstance(message, discord.Message)
+        voice_channel = await self.get_voice_channel(message)
+        if voice_channel is not None:
+            for voice_client in self.voice_clients:
+                if voice_client.guild == message.guild:
+                    logging.info("Self in voice channel")
+                    return voice_client
+            return await voice_channel.connect_deaf()
+        return None
+
     async def play(self, message, command_content):
         """
         Play URL or first search term from command_content in the author's voice channel
         """
-        video_metadata = None
+        voice_channel = await self.get_voice_channel(message)
+        if voice_channel is None:
+            # Exit early if user is not connected
+            return
 
+        video_metadata = None
         if self.url_regex.match(command_content):
             video_metadata = pafy.new(command_content)
         else:
             search_result = VideosSearch(command_content).result()
             video_metadata = pafy.new(search_result["result"][0]["id"])
 
-        logging.info("Pafy found %s", self.user)
-        logging.info(str(video_metadata))
-        audio_url = video_metadata.getbestaudio().url
-
-        if self.voice_client is None:
-            self.voice_client = await message.author.voice.channel.connect()
-            await self.voice_client.guild.change_voice_state(
-                channel=message.author.voice.channel, self_deaf=True
+        voice_client = await voice_channel.connect_deaf()
+        if voice_client.is_playing():
+            await message.channel.send(
```

Comment:
Maybe move this message sending to after the song has actually been added to the queue. Same for "Now playing" message.